scheduler/pkg/kafka/gateway/infer.go

```diff
@@ -296,9 +332,10 @@ func (kc *InferKafkaHandler) AddModel(modelName string) error {
 	return nil
 }
 
-func (kc *InferKafkaHandler) RemoveModel(modelName string) error {
+func (kc *InferKafkaHandler) RemoveModel(modelName string, cleanTopicsOnDeletion bool) error {
```

Comment:
RemoveModel is called any time we receive a status update from the scheduler and that reports that the model [has zero available replicas](https://github.com/SeldonIO/seldon-core/blob/a6deb439330b97e486d91d89609690cc2cab470f/scheduler/pkg/kafka/gateway/client.go#L176). This might be a bit eager: I suspect that if the scheduler gets disconnected from inference server agents (network partition), or (easier to test) if a model gets scaled down to 0 replicas, then the topics get deleted. We would want to avoid that.  We should also test what happens if a model gets updated (new version of the CR applied): are the topics deleted & recreated? We would also want to avoid that scenario (loosing topic data on model version update)

Would you be able to test those scenarios?

Reply:
The scenarios above seem to be covered by [this](https://github.com/SeldonIO/seldon-core/blob/02a7a8e6457285d53cc7700b207f815e6e90603f/scheduler/pkg/kafka/gateway/client.go#L162-L170).